### ISSUE TITLE
Check parsing of dateTimeOffsets (should we use ticks?)

### ISSUE THREAD

**Issue by mmsmits** [FirelyTeam/firely-net-sdk#2281]
Serialization roundtripping of instance datatypes doesn't work yet.

Example

Input:
```<timestamp value="2020-04-17T10:24:13.1882432-05:00"/>```

Output:
```<timestamp value="2020-04-17T10:24:13.188-05:00"/>```

I think the error occurs when parsing the string values to DateTimeOffsets using
DateTime.ToDateTimeOffset() in DateTime.cs

Unit test to recreate this issue:

input: ```2020-04-17T10:24:13.1882432-05:00```
output: ```2020-04-17T10:24:13.188-05:00```

```

        [TestMethod]
        public void DateTimeOffsetTest()
        {
            var input = @"2020-04-17T10:24:13.1882432-05:00";
            var datetime = ElementModel.Types.DateTime.Parse(input);
            var offset = datetime.ToDateTimeOffset(TimeSpan.Zero);
            var output = ElementModel.Types.DateTime.FormatDateTimeOffset(offset);

            output.Should().Be(input);
        }
```

**Comment by ewoutkramer** [comment]
Incidentally, I hit this same error when working on #2541, so this has now been fixed.

The cause of the error is that there is no way to get the *fractions* from a DateTimeOffset!  You can ask for the Millis and the Micros and the Nanoseconds instead - but then again there is no constructor for DateTimeOffset that takes Micros, so when constructing a new DTO based on Millis, you'll lose some of this information.

Indeed, the only way to create a new DateTimeOffset based on another one, is by doing new DateTimeOffset(old.Ticks);

This fixed the problem you provided the test for.

**Comment by ewoutkramer** [comment]
Fixed by #2541.
